rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,166 @@
+
+The `https://github.com/linebender/xilem` repository has the following crates in its `crates/` directory:
+
+- `xilem_core/`
+- `xilem_web/`
+- `xilem_native/`
+- `masonry/`
+- `xilem/`, which would only be a placeholder crate pointing to `xilem_web/` or `xilem_native/`.
+
+
+## Implementation strategy
+
+The implementation will require several steps:
+
+- Refactoring Masonry to use the same dependencies Xilem currently uses, especially Glazier, Vello, and Parley insted of druid-shell and Piet.
+- Integrating AccessKit support.
+- Moving the Masonry codebase to the Xilem repository, ideally in a way that preserves commit history.
+- Porting the list of Github Issues in the masonry-rs repository to the Xilem repository.
+- Creating a new `xilem_native` crate on parity with the current Xilem crate.
+- Removing the current xilem crate.
+- Publishing both `xilem_native` and the placeholder `xilem` crate on crates.io.
+- Sharing ownership of the crates with Raph Levien.
+
+
+## Drawbacks
+
+### Layer separation
+
+This move decouples the reactive layer of xilem from its platform layer by putting them in separate crates.
+
+This means some features in `xilem_native` may be harder to implement because they won't have access to the backend's private code.
+
+### Jumping to Masonry
+
+By using the Masonry codebase, the Xilem project will move towards concept Masonry was implementing that Xilem maintainers weren't necessarily considering for Xilem.
+
+Some PRs intended for Xilem may become obsolete once the move to Masonry is complete, because Masonry will go in a different direction from the one this PRs were made for.
+
+
+## Rationale and alternatives
+
+### Separating at the crate level vs the repository level
+
+The initial suggestion was to use Masonry as it currently exists, as a separate repository.
+
+However, people have brought up that it would make the Xilem development process a lot more complex. Features that reached across crates (`xilem_core`, `xilem_native`, `masonry`) would need to be rolled out at coordinated pull requests between repositories.
+
+### Separating at the crate level vs the module level
+
+Masonry and Xilem's code could simply be merged in a single crate, under different modules.
+
+The consensus was mostly that having Masonry as a separate crate would help keep Xilem "honest" and enforce separation of concerns.
+
+More importantly, Masonry was built as a potential foundation for *multiple* native GUI crates, and it would be valuable to conserve that; therefore, while Masonry would still depend on crates from the Linebender ecosystem, it shouldn't be too tied to Xilem.
+
+### Porting Xilem to Masonry vs Masonry to Xilem
+
+In any merge between two projects, one has to be the base the other is grafted on.
+
+This RFC was written under the assumption Xilem would move towards Masonry, but the opposite could be possible.
+
+That said, the Motivation section explains why Masonry is a healthier codebase to start from.
```

Comment:
I expect to just take the best code in one or the other and just use that (and otherwise merge the good bits of the other).
Xilem is the frontfacing layer, so it depends on masonry otherwise?

Reply:
Xilem as it currently is has both frontend and backend code. The question how much of that backend code we want to build from vs throwing it away and starting from Masonry's code.

Reply:
Yes and as I've written, I think the good parts can just be merged, I don't think it's an either/or thing